actionpack/lib/action_dispatch/middleware/ssl.rb

```diff
@@ -80,9 +87,8 @@ def call(env)
           flag_cookies_as_secure! headers if @secure_cookies
         end
       else
-        return redirect_to_https request if @redirect
+        return redirect_to_https request if @constrain_to.call(request)
         @app.call(env)
-      end
```

Comment:
This was an accidental push. I had some merge conflicts and messed up when I removed the git merge diff from the file.